app/serializers/dist/json_serializer.rb

```diff
@@ -70,5 +72,61 @@ def serialize_nested(connection)
         name: connection.name,
       }
     end
+
+    def build_mapping_blocks(mapping_rules)
+      all_mapping_blocks = []
+      puts "Generating mappings ..."
+      mapping_count = 0
+      Integration.all.each do |integration|
+        [true, false].each do |from_shopify|
```

Comment:
```suggestion
      mapping_rules = Integration.all.flat_map do |integration|
        [true, false].filter_map do |from_shopify|
          rules = mapping_rules.where(integration:, from_shopify:)
          rules if rule.any?
        end
      end
```

Reply:
Mostly, if you decompose this it'll be easier to read with each piece being more focused. Also makes it simpler to use map methods, which are nice that each with manual tracking arrays 😁